api/core/v1alpha1/digest.go

```diff
@@ -21,27 +20,21 @@ func init() {
 	}
 }
 
-func (d *Digest) ToString() string {
-	return fmt.Sprintf("%s:%x", DigestType_name[int32(d.GetType())], d.GetValue())
+func (d *Digest) Encode() string {
+	return fmt.Sprintf("%s:%s", DigestType_name[int32(d.GetType())], d.GetValue())
 }
 
-func (d *Digest) FromString(str string) error {
+func (d *Digest) Decode(str string) error {
 	// parse and validate digest
 	parts := strings.Split(str, ":")
 	if len(parts) != 2 {
 		return fmt.Errorf("digest parts not found")
 	}
 
-	// extract signature
-	digestValue, err := hex.DecodeString(parts[1])
-	if err != nil {
-		return fmt.Errorf("failed to decode digest: %w", err)
-	}
-
 	// update digest
 	*d = Digest{
```

Comment:
i would still validate the digest value itself with the following:
- is the value read a valid HEX string
- is the value length valid for a given digest type (e.g. sha256 should have 64 chars)